src/viztracer/modules/snaptrace.c

```diff
@@ -329,8 +332,22 @@ snaptrace_tracefunc(PyObject* obj, PyFrameObject* frame, int what, PyObject* arg
         if (CHECK_FLAG(self->check_flags, SNAPTRACE_LOG_ASYNC)) {
             if (info->curr_task == NULL) {
                 if (is_python && is_call && (frame->f_code->co_flags & CO_COROUTINE) != 0) {
+                    PyObject* curr_task = Py_None;
                     info->paused = 1;
-                    PyObject* curr_task = PyObject_CallObject(asyncio_tasks_current_task, NULL);
+                    for (int i = 0; i < sizeof(curr_task_getters)/sizeof(curr_task_getters[0]); i++) {
+                        if (curr_task_getters[i] == NULL) {
+                            continue;
+                        }
+                        curr_task = PyObject_CallObject(curr_task_getters[i], NULL);
+                        if (!curr_task) {  // RuntimeError, probably
+                            PyErr_Clear();
+                            curr_task = Py_None;
+                            continue;
+                        }
+                        if (curr_task != Py_None) {
+                            break;
+                        }
+                    }
```

Comment:
This part is basically what I imagined. Let's clear up some logic.

Line 338, I think for shorter loop body like this, it would be better to write
```
if (curr_task_getters[i] != NULL) {
    // stuff
}
```
than using ``continue``.

line 342, I think ``curr_task`` should either be some useful value, or ``NULL`` with a run-time error? So wouldn't it be better here to have
```
if (!curr_task) {
    PyErr_clear();
    curr_task = Py_None;
    // remove continue here
} else {
    break;
}
```

Actually it might be more logical to have the "correct" case first, do ``break`` in ``if``, and clear errors in ``else``.

Reply:
> line 342, I think `curr_task` should either be some useful value, or `NULL` with a run-time error? So wouldn't it be better here to have

That's not quite right, `asyncio` can return `None` if there is a loop but no task.  I'm not exactly sure when that can happen but we should account for that and not break and try the next task getter.